### ISSUE TITLE
Question regarding to Eq(8) in your paper

### ISSUE THREAD

**Issue by JixiangChen-Jimmy** [YongHyun-Ahn/LINe-Out-of-Distribution-Detection-by-Leveraging-Important-Neurons#1]
Dear authors, thanks for the interesting work! Nevertheless, I am confused about the final OOD score calculation. According to Eq(8). in your paper, for each in-distribution class, a logit of dimension L (i.e., ID class number) is obtained for each class l, as a pruned weight is obtain for each class. So what is the next step to calculate the final OOD score? As shown in Fig.2, LINe seems to leverage the energy score function, is that so? And how to incoporate the L class-specific logits for OOD score calculation? I did not find further explanation in the paper. Looking forward to your reply.

**Comment by YongHyun-Ahn** [comment]
Dear @JixiangChen-Jimmy, Thank you for your comment. 
In test time(i.e., inference time) LINe leverages the predicted label of the sample. 
LINe only uses two masks for each sample (one for AP and the other for WP).
So for the penultimate layer output h(x) and predicted class as '_l_'. 
these are the following steps to calculate the final OOD score. 

1. get h(x) and '_l_' from model.
2. Activation clipping (AC)
3. Activation Pruning (AP) - with one activation mask for class '_l_'  (which is denoted as '_m_l_' in Figure 2)
4. Weight Pruning (WP) - with one weight mask for class '_l_' (which is denoted as 'M^l_w' in Figure 2)
5. Matrix multiplication between Masked feature and Masked weight 
6. Get the energy score from the calculated logit.

You can find details for 3,4,5 in _/models/route.py_, and _/util/score.py_ for 6.
We hope this comment solves your question.
